document/js-api/index.bs

```diff
@@ -1148,9 +1148,9 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
     1. Return [=f64.const=] |f64|.
 1. If |type| is of the form [=ref=] |null| |heaptype|,
     1. If [=match_valtype=](|type|, [=ref=] |null| [=heap-type/extern=]),
-        1. Let |r| be [=ToWebAssemblyValue=](|v|, [=ref=] |null| [=heap-type/any=]).
-        1. Return [=ref.extern=] |r|.
-    1. If |v| is null,
+        1. Let |rr| be [=ToWebAssemblyValue=](|v|, [=ref=] |null| [=heap-type/any=]).
+        1. Let |r| be [=ref.extern=] |rr|.
+    1. Else if |v| is null,
```

Comment:
I think the null case and extern case may need to be swapped. If `v` is null and `type` is `(ref null noextern)` then this current conditional will pick the extern branch and set `r = ref.extern (ref.null noextern)` which has type `extern`. Then the subtype check fails.

Reply:
Yes, you're right. I also moved the func case back to the beginning, since it is more similar to extern now.